### ISSUE TITLE
Customize dataset format.

### ISSUE THREAD

**Issue by yinheyanxian** [inakierregueab/YOLO11-JDE#23]
Hello. Firstly, thank you for your wonderful work.

I want to replicate your work on the Visdrone2019-MOT dataset, but I don't know how to organize the dataset format.

Based on your previous answer, I understand that the training set is in pure YOLO format, where one folder stores images and another folder stores the corresponding YOLO txt annotations (class x_centery_center width height). The validation set needs to add a track ID after the YOLO annotation (class x_centery_center width height track_id).

But how should the validation set be organized? Is it also a folder for storing images and another folder for storing corresponding labels?

The validation set of Visdrone2019-MOT has 7 sequences. Do I need to put all 7 sequences together or evaluate them separately?

The following figure shows the seven sequences and corresponding labels of the Visdrone2019-MOT validation set.

<img width="1738" height="428" alt="Image" src="https://github.com/user-attachments/assets/96c87c04-4fd1-44b8-b7bb-f59c664292ee" />

**Comment by inakierregueab** [comment]
Hi, thank you:)

You are on the right path! The validation set should follow the same format as the training set but including the track_id column so that it is possible to eval the re-id capabilities of the model. If you have 7 video, I recommend you extract frames of every video at the desired sampling rate, matching the videos FPS or a bit more sparse, depending on your needs and objetives. Extract all frames and name them accordingly, something like {video_name}_{frame_number}.png. I suggest that in the same loop you get the corresponding annotations for that frame and save them in a txt with the same filename as the image. After this process you will have N video frames in png and N txt files, one per every frame. 

Regarding the folder structure: dataset_name/{images/labels}/{train/val/test}.

Btw, it will be nice to see how re-id performs on visdrone, since most objects are tiny, very challenging. Training and inference image size can be a key parameter to experiment with.

**Comment by yinheyanxian** [comment]
> Hi, thank you:)
> 
> You are on the right path! The validation set should follow the same format as the training set but including the track_id column so that it is possible to eval the re-id capabilities of the model. If you have 7 video, I recommend you extract frames of every video at the desired sampling rate, matching the videos FPS or a bit more sparse, depending on your needs and objetives. Extract all frames and name them accordingly, something like {video_name}_{frame_number}.png. I suggest that in the same loop you get the corresponding annotations for that frame and save them in a txt with the same filename as the image. After this process you will have N video frames in png and N txt files, one per every frame.
> 
> Regarding the folder structure: dataset_name/{images/labels}/{train/val/test}.
> 
> Btw, it will be nice to see how re-id performs on visdrone, since most objects are tiny, very challenging. Training and inference image size can be a key parameter to experiment with.

Thank you for your answer!
I have now implemented the data conversion of Visdrone2019-MOT and successfully trained it.

<img width="2422" height="914" alt="Image" src="https://github.com/user-attachments/assets/d052ad88-deaf-4557-a5a0-908a45ec844e" />

However, I have discovered a new issue.

At first, I trained a lightweight YOLO11n-JDE on a server with 32GB of RAM and 4080GPU, with an epoch of 8. During the second epoch evaluation, the training process was always killed by the system. According to the previous question (# 21), it seems that up to 100GB of RAM will be occupied during the verification period.

The RAM required for evaluation after each epoch training seems to be very large.

Now, I am training it on a server with 120GB of RAM and 4090GPU. Although it can be trained normally, the evaluation time for each epoch is very long, far exceeding the evaluation time of general object detection algorithms.

Is there a way to improve this problem?

Thanks in advance!

**Comment by inakierregueab** [comment]
That might happen because your val set is too big. Try with a smaller one that is still representative.

**Comment by yinheyanxian** [comment]
> That might happen because your val set is too big. Try with a smaller one that is still representative.

Hi.
I have successfully trained and validated it now, but when evaluating tracking performance, the following error occurred:

<img width="2376" height="80" alt="Image" src="https://github.com/user-attachments/assets/31c02659-6ba3-4d55-9b5f-7a8133813ddc" />

It seems that additional data labels are needed to evaluate tracking performance. 

<img width="2354" height="898" alt="Image" src="https://github.com/user-attachments/assets/04fd65ed-d23c-459b-8fbb-1f5a45c3c39e" />

Can you tell me what this is? How to organize labels that can be used for evaluation.
Thank you!

**Comment by inakierregueab** [comment]
Hi, that is a callback function. It is used to eval the Re-id task during validation while training. This callback performs inference with the model over the validation set and then uses the tracker to perform the ID matching across consecutive frames and finally evaluates using the TrackEval evaluator (https://github.com/JonathonLuiten/TrackEval). If you want to test on your own custom data, you will also have to convert your evaluation data to the oficial MOT format, because that is the format used by the evaluator too. 

\<frame\>, \<id\>, \<bb_left\>, \<bb_top\>, \<bb_width\>, \<bb_height\>, \<conf\>, \<x\>, \<y\>, \<z\>


The last 3 columns can be set to -1. 

You can find more info here: https://motchallenge.net/instructions/ and in the oficial repo of TrackEval. You only need to build the gt txt file for your dataset and update the seqmap_file path correctly in the code to point to your gt data.

**Comment by zhangguozheng-1** [comment]
Could you show me an example of each row of data in the val dataset? I'm trying to train on my own dataset now, but I haven't succeeded yet，Thank you!
